README
GOOGLE CHECKOUT PLUGIN FOR ZEN CART v1.5.1 - 2012-03-09

INTRODUCTION
============
The Google Checkout module for Zen Cart adds Google Checkout as a Checkout 
 Module within Zen Cart.
This will allow merchants using Zen Cart to offer Google Checkout as an 
 alternate checkout method.
The plugin provides Level 2 integration of Google Checkout with Zen Cart.

Plugin features include:
1. Posting shopping carts to Google Checkout
2. Shipping support with Merchant Calculated Shipping Rates and 
    Carrier Calculation Shipping
3. Tax support
4. User and order updates within Zen Cart
5. Order processing using Zen Cart Admin UI 
6. Order Totals Support
7. Digital Delivery support

REQUIREMENTS
============
Zen Cart v1.3.8, v1.3.9a-h and v1.5.0
PHP3 or PHP4 or PHP5 with cURL(libcurl) installed and enabled

INSTALLATION NOTES
==================
1. Follow instructions contained in the INSTALLATION file.
2. Verify the installation from the Admin site and selecting MODULES->PAYMENTS 
    and checking if Google Checkout is listed as a payment option.
3. Set the file attribute to 777 for /googlecheckout/logs/response_error.log and 
    /googlecheckout/logs/response_message.log files.
4. Go to http://<url-site-url>/googlecheckout/responsehandler.php
    If you get a 'Invalid or not supported Message', 'Fail to get HTTP Authentication',
    or a request for User and Password, go to the next section.
    If you get any errors,  you must correct all errors before proceeding.  
    Refer to the troubleshooting section below or go to the support forum for help.


SETUP ON ADMIN UI
=================
Select and install the Google Checkout payment module. The following are some 
of the fields you can update:

0. Installed GC module version
1. Enable/Disable: Enable this to use Google Checkout for your site.

2. Operation Mode: Test your site with Google Checkout's sandbox server before 
   migrating to production. You will need it signup for a separate Google Checkout 
   sandbox account at http://sandbox.google.com/checkout/sell. Your sandbox account
   will have a different Merchant ID and Merchant Key. When you are ready to run 
   against the production server, remember to update your merchant ID and key 
   when migrating.
   
3. Merchant ID and Merchant Key:(Mandatory) If any of these are not set and the 
   module is enabled, a disabled (gray) Checkout button appears on the Checkout 
   page. Set these values from your seller Google account under the 
   Settings->Integration tab. Separate Sandbox and Production.
   
4. .htaccess Basic Authentication Mode with PHP over CGI? If your site is 
   installed on a PHP CGI you must disable Basic Authentication over PHP. 
   To avoid spoofed messages (only if this feature is enabled) reaching 
   responsehandler.php, set the .htaccess file with the script linked 
   (http://your-site/admin/htaccess.php). 
   Set permission 777 for http://your-site/googlecheckout/ before running 
   the script. Remember to turn back permissions after creating the files.
   
5. Merchant Calculation Mode of Operation: Sets Merchant calculation URL for 
   Sandbox environment. Could be HTTP or HTTPS. (Checkout production environment 
   always requires HTTPS.)
   
6. Disable Google Checkout for Virtual Goods?: This configuration is enabled and
    there is any virtual good in the cart the Google Checkout button will be
    shown disabled.
    (double check http://checkout.google.com/seller/policies.html#4)
    
7. Allow US PO BOX shipping. Setted to false, you won't ship to any PO address
    in the US.

8. Default Values for Real Time Shipping Rates: Set your default values for 
   all merchant calculated shipping rates. This values will be used if for any 
   reason Google Checkout cannot reach your API callback to calculate the 
   shipping price.

9. GoogleCheckout Carrier Calculated Shipping. Set if you want or not to use CCS.
    Note that if you enable CCS, all Merchant Calculation Modules will be IGNORED
    for GC orders. Only Flat Rate shippings will be included with CCS.

10. Carrier Calculater Shipping Configuration. Set Default Values, Fix and 
    Variable charge for each specific CCS method. 
    The DV contains the default shipping cost for a carrier-calculated-shipping 
     option. If Google is unable to obtain the carrier's shipping rate for a 
     shipping option, the buyer will still have the option of selecting that 
     shipping option and paying the DV value to have the order shipped.
     http://code.google.com/apis/checkout/developer/Google_Checkout_XML_API_Carrier_Calculated_Shipping.html#tag_price
    The Fix value allows you to specify a fixed charge that will be added to the
     total cost of an order.
     http://code.google.com/apis/checkout/developer/Google_Checkout_XML_API_Carrier_Calculated_Shipping.html#tag_additional-fixed-charge
    The Variable Charge pecifies a percentage amount by which a carrier-calculated
     shipping rate will be adjusted. The tag's value may be positive or negative.
     http://code.google.com/apis/checkout/developer/Google_Checkout_XML_API_Carrier_Calculated_Shipping.html#tag_additional-variable-charge-percent
    
    Set Def. Value to 0 to disable the method

10. Rounding Policy Mode and Rounding Policy Rule: Determines how Google Checkout
     will do rounding in prices.
     US default: rounding rule TOTAL, rounding mode is HALF_EVEN
     UK default: rounding rule PER_LINE, rounding mode is HALF_UP

     More info:
     http://code.google.com/apis/checkout/developer/Google_Checkout_Rounding_Policy.html

11. Cart Expiration Time: if different from NONE, this postive integrer will set 
     an expiration time from the creation of the GC button to X minutes. Where X
     is the value set. This will prevent buyers to use carts that may be deprecated
     or with wrong settings. Default is NONE. Take in care that the time in your
     server may not be syncronized with GC servers, so if you set short expiration
     times this may not be so accurate.     

12. Also send Notification with ZenCart, if enabled, will send an email using the 
     ZC internal email system if the comment in an order is larger than 254 
     chars, limit for a google send message. If this happens a warning will be 
     shown in the Admin UI. It will also send emails to the merchant account 
     when orders states are changed in the Admin UI.

13. Google Analytics Id: Add google analytics to your e-commerce. Now there is a 
     feature in GA to integrate easily with any e-commerce with GoogleCheckout.
     More info: See below "Enabling E-Commerce Reporting for Google Analytics".

14. 3rd Party Tracking: Do you want to integrate the module 3rd party tracking? 
     Add the tracker URL, NONE to disable.
    More info:
    http://code.google.com/apis/checkout/developer/checkout_pixel_tracking.html

15. Google Checkout restricted product categories: Insert here the Ids of all the 
     product categories that you want to exclude from the Google Checkout. 
     Separate them using commas (ie. "1, 3,56 , 32").
     Double check Google Policy:
     http://checkout.google.com/support/sell/bin/answer.py?answer=46174&topic=8681
     http://checkout.google.com/seller/policies.html#4

16. Continue shopping URL: The URL customers will be redirected to if they 
     follow the link back to your site after checkout. 
     (http://your-site/zencart_dir/index.php?main_page=<input data>)
     Note:Use GC_return for special page that will show all the purchased items 
      with Google Checkout

Your Google Checkout setup page is correct if, upon viewing it, a non-disabled 
Google Checkout button appears. Double check the INSTALLATION file for more info

Enabling E-Commerce Reporting for Google Analytics
==================================================
To track Google Checkout orders, you must enable e-commerce reporting for your 
website in your Google Analytics account. The following steps explain how you 
enable e-commerce reporting for your website:

   1. Log in to your Google Analytics account.
   2. Click the Edit link next to the profile you want to enable. This link 
      appears in the Settings column.
   3. On the Profile Settings page, click the Edit link in the Main Website 
      Profile Information box.
   4. Change the selected E-Commerce Website radio button from No to Yes.
More info: 
http://code.google.com/apis/checkout/developer/checkout_analytics_integration.html

Note for 3rd party tracking: Actual configuration supports just one 3rd party 
 tracking Co. And some modification maybe needed to do in the code for 
 specific trakers.
 Read:
 http://code.google.com/apis/checkout/developer/checkout_pixel_tracking.html
 And change the code here:
  googlecheckout/gcheckout.php
 Maping works this way:
  $tracking_attr_types = array(
                              'GC_attr_type1' => '3rd_attr_name1',
                              'GC_attr_type2' => '3rd_attr_name2',
                              ...
                              );
  Will be traduced to:
  <parameters>
    <url-parameter name="3rd_attr_name1" type="GC_attr_type1"/>
    <url-parameter name="3rd_attr_name2" type="GC_attr_type2"/>
    ...
  </parameters>
  
DIGITAL DELIVERY
================
All products marked as maked as "Product is Virtual" will be included in the 
 digital delivery API. This means that at the end of the GC transaction a link
 to the ZC checkout_success page will be shown as well as a description of the 
 product.
If the whole cart is virtual, no shipping to: will be shown in the GC Place 
 Order Page. 
Note: If the cart is virtual but has some custom order_totals (low_orderFee, 
 group_discount, etc), a notice saying an "Email will sent" will be show for each
 order total in the GC confirmation page. This is a limitation of GC API for
 supporting custom OT and digital delivery.

MERCHANT CALCULATED SHIPPING
============================
In order to use this module you must have some Real Time Shipping provider,
 such as USPS or FedEx. This Module must be activated and configured in
 Modules->Shipping. For each enabled module you'll have to set the default
 values in the Google Checkout Admin UI.
This Value will be used if for any reason Google Checkout cannot reach your
 API callback to calculate the shipping price. 

The available shipping methods for each shipping provider must be configured
 in /googlecheckout/shipping_methods.php in the mc_shipping_methods
 variable. If you want to disable one or more methods, just comment them out.
Be aware that if you mix flat rate and real time rates, both will be taken
 as merchant-calculated-shipping. 

Script to create new shipping methods
	http://your-site/zencart_dir/googlecheckout/shipping_generator
More Info: 
	http://your-site/zencart_dir/googlecheckout/shipping_generator/README

CARRIER CALCULATED SHIPPING
===========================
This feature allows merchants to get real time quotes without any effort. All
 calculation are done by the GC servers. You only need to specify the shipping
 methods you want to provide. 
 Google supports the carrier-calculated shipping feature for three carriers: 
 FedEx, UPS and the U.S. Postal Service (USPS)

You cannot offer carrier-calculated shipping methods and merchant-calculated
 shipping methods for the same order. However, you can offer carrier-calculated
 shipping methods and still use the Merchant Calculations API to calculate taxes 
 and adjustments for coupons and gift certificates.
 
Note: Only US Domestic shipping address are allowed right now. If shipping 
       address is Int'l only Flat Rate shippings will be shown.

More info:
 http://code.google.com/apis/checkout/developer/Google_Checkout_XML_API_Carrier_Calculated_Shipping.html#Process

COUPONS
=======
You can now use coupons with this module. Just add coupons in the Admin UI,
 and the user can addlogs/ them in his order. This is restinged to just one
 coupon per order as Zen cart standards, aldo GC supports many. 

TRACKING USERS AND ORDERS
=========================
In order to provide this support (as required for Level 2 integration), update
 the API callback request field in the seller account to 
 https://<url-site-url>/googlecheckout/responsehandler.php . 
 Note that the production Checkout server requires an SSL callback site.
The Sandbox server accepts non SSL callback URLs.

Within the Google Checkout panel, go to Settings > Integration. Set the
Callback contents to Notification as XML, rather than Notification Serial
Number.

View your Google Checkout customers and their orders in the Reports tab. 
For each order, the default starting state is Google New. 
GC Orders States in ZC Admin

(the instructions below assume you have enabled "Automatically authorize and
charge the buyer's credit card" in your GC merchant account Settings >
Preferences).

1. When an order for a regular (non-download) product is submitted  through
GC the order state in ZC admin will be auto set to "Google_Processing". When
following-up on (shipping) these orders you will need to change the order
status to "Google_Shipped".

2. When an order for a download product is submitted via GC, the order
status in ZC admin will be auto set to "Google_Shipped" (even if customer
has attempted to download or not). When following-up on download orders keep
the
status "Google_Shipped", no need to change the state.

3.. Refunding/canceling "total" GC orders works from within ZC admin.

a. If you have not yet changed a non-download order from "Google_Processing"
to "Google_Shipped" you can directly cancel/refund the TOTAL order by
selecting "Google_Canceled" (but do NOT set to "Google_Refunded").

Changing the state to "Google_Cancelled" will auto refund 100% of
transaction AND cancel the order at the same time. So there is no need to
login to GC Merchant account to process the TOTAL refund.

b. Also after an item is shipped you can refund the TOTAL amount of order by
 choosing "Google_Canceled" (but as stated above do NOT set to
"Google_Refunded").

c. However if you want to only refund a PORTION of the order amount (for
example if customer ordered two items and want to refund amount for only one
item) then you will need to log in to your GC Merchant amount and process
the partial refund. (Then Google will automatically change the order status
in ZC admin to "Google_Refunded" or "Google_Shipped and
Refunded", and you won't have to change the state in ZC admin.

d. Understanding the above then you should not ever need to apply
"Google_Refunded" or "Google_Shipped and Refunded". These states are usually
reserved for Google use only.

You can add a Tracking Number in the state change from GoogleProcessing to
GoogleShipped
A text field and a combo with the shipping providers will be show when the order
is in the GoogleProcessing state.

Any comments added during state change will be sent to the buyer account page
 if you have selected the Append Comments option.

All statechanges are added as notes in the Admin UI
All request and response messages will be logged to the file
googlecheckout/logs/response_message.log.

Refunds are added as new order totals in each order as well as part of the
history of the order.

The same for cancellations

Check states.jpg for the states machine diagram. 

PROJECT HOME
============
To check out the latest release or to report issues, go to
 https://github.com/mattclements/Google-Checkout-for-Zencart


GROUP DISCUSSIONS
=================
To meet other developers and merchants who have integrated Google Checkout
 with Zen Cart and to discuss any topics related to integration of Google 
 Checkout with Zen Cart, go to 
 https://github.com/mattclements/Google-Checkout-for-Zencart/issues

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
MOST COMMON MISTAKES
====================
1. Make sure you set the file attribute to 777 for 
    /googlecheckout/logs/response_error.log and /googlecheckout/logs/response_message.log files.
2. Set your Google callback url to https://<url-site-url>/googlecheckout/responsehandler.php
   In Sandbox, HTTPS is not required.

CHANGELOG
CHANGE LOG
==========

09/26/2006 v1.0  (Google Checkout team)
	- Initial release. 
	
09/27/2006 v1.0  (Google Checkout team)
	- Updated the original module to run on either PHP4 or PHP5. 
	
09/28/2006 v1.0a (DrByte)
	- Code separated into components following Zen Cart standards.
	- Zipped for easier implementation.
	- Readme updated. 
		   
10/05/2006 v1.0.1 (Google Checkout team)
	- Step-by-step installation instructions included.   
	
10/14/2006 v1.0.2 (colosports)
	- Fixed languages_id bug.
	- Added return links from Google Checkout Cart.
	- Added customers ip-address & phone number.
	- Added getallheaders() funtion.
	- Fixed zone id bug.
	- Fix address book bug.
	- Added Google link and Order No: to Admin page.
	- Fixed quantity bug.
	- Fixed product attribute bug.
	- Added product attribute in Google Checkout cart. 
	- Updated README file. 

10/14/2006 v1.0.2c (colosports)
	- Fixed change order status on Admin.
	- Change time to reflect time zone on Admin. 
	- Separated full name to first and last name.
	- Updated README file. 	

10/19/2006 v1.0.3 (Google Checkout team)
	- Fixed Notify Customer checkbox bug

11/16/2006 v1.0.4 (colosports)
	- Fixed order status bug.
	- Updated README file and INSTALLATION file. 	
11/28/2006 v1.0.5 (ropu)
   - Add support for Merchant Calculated Shipping Rates.
   - Fixed minor bugs in responsehandler.php and orders.php files
   - Change the XML parser and builder
   - Removed getallheader() function
   - Fixed wrong Qty in Admin UI
   - Fixed modules not saving their settings
   - Fixed Notify Customer option
   - Add state change, Risk and charge notification into the Admin UI
   - Add support for coupons
   - Fix Shopping cart not obtained from session. See TROUBLE SHOOTING.	
01/12/2007 v1.0.5b1 (ropu)
   - Fix <tax-table-selector> strict validation.   

02/26/2007 v1.2 (ropu)
   - Tested on Zencart v1.3.7
   - Add multisocket feature for merchant-calculations (alfa)(optional)
   - Different algorithm to retrieve quotes
   - Add Google Analytics Support
   - Add support for PHP CGI installations
   - Add user and password for Google Checkout buyers
   - Items retrieved from Merchant-private-item-data instead of session.
   - Many bug fixes
   
03/05/2007 v1.2 RC4 (ropu)
   - Fix gray button when Tax Class selected bug (Thx BlessIsaacola)

03/30/2007 v1.3RC1 (ropu)
   - Add tracking data to the Zencart Admin UI Orders
   - Fixed SSL issue with Google Analytics feature
   - International Shipping Features
   - Restricting Shipping Options for Post Office (P.O.) Box Addresses feature
   - International Tax Features
   - Selecting a Rounding Policy for Tax Calculations
   - Fixed Tax for zones
   - Fixed Tax for products
   - Added support for All Areas Zones
   - Coupon disabled bug
   - Added coupon zone restriction validation (Based on delivery address, not billing)
   - Added coupon product/category restriction validation
   - Add a configuration to disable Google Checkout Button when are virtual good in the cart
   		(double check http://checkout.google.com/seller/policies.html#4)
   - Disable multisocket Option :(
   
04/16/2007 v1.3RC2 (ropu)
   - Support for UK merchants 
     http://code.google.com/apis/checkout/developer/release_notes.html (apr13)
   - Updated Templates to Zencart v1.3.7
   - Remove tpl_checkout_payment_default.php
   - Docs update
   - Minor bug fix
   - Added Shipping Generator Tool in the stardard package (Optional)
   - Flat rate shipping bug fixed
   - Improved tax and shipping restrictions
   - Added configurable Continue shopping URL.
   - Fix Warning: Invalid argument supplied for foreach() in 
   		/home/public_html/googlecheckout/gcheckout.php on line 390
   - Fixed shipping not showing correctly in the orders page
   
06/11/2007 v1.4 alpha (ropu)
   - Refactor all code using PHP Sample Code as the base library (under googlecheckout/library)
     - Handles Logging, added log levels (error, request, response)
       - Configuration from UI (comming soon), needed??
     - Handles Basic authentication, add support for IIS (See TROUBLESHOOTING)
     - Handles all XML parsing and building (New xml parsed used)
   - Added OT support (low order fees, group discounts and any other custom OT module)
   - Fixed Split login template
   - Add GC button in timeOut login page
   - Remove GC from step 2
   - Fixed freeshipper bug
   - Add warning messages when GC is disabled.
   - Add stack message in Admin UI with GC orders state changes and errors
   - Added configuration to also use Zencarts email system for GC orders
   - Added a fix to avoid regular checkout with GC if no template is applied (or wrongly applied)
   - Added a metric tester for shipping times feedback (optional)
   - Added 3rd party tracking
   - Added a coustom Continue shopping url with GC bought items.
   - Moved logs to a .htacces protected dir (googlecheckout/logs)
   - Added Correct HTTP status (2xx, 4xx, 5xx) return when errors occurrs with the correct message
   - All strings are defined in the language include (only english right now)
   - Use DEFAULT_CURRENCY as the currency in GC cart, not buyers session one
   - Added version number to all files
   - Applied XSS patch to orders.php
   - Slipt README and CHANGELOG
 	 
 	 
07/07/2007 v1.4alpha3 (ropu)
    - Refactor new-order-notification handling using more Zencart functions
    - ZC DB 'orders' table fields 'order_total' and 'order_tax' are written.
    - Merged the GC return page with the zencart default one
    - Instant Downloads Support.
    - Support Google Refunds and Cancellations
    - GC Invoicing issue, fixed, not tested
    - If the customers e-mail address already exists in Zen then add the
        order placed to their account instead of creating a new customer.
    - If the GC exits in ZC but uses a different email then add the
        order placed to their account instead of creating a new customer.
    - New customers created are sent details of their account that

googlecheckout/shipping_methods.php
<?php
/*
 **GOOGLE CHECKOUT ** v1.5.0
 * @version $Id: shipping_methods.php 5852 2007-12-14 14:58:57Z ropu $
 * 
 */
  // this are all the available methods for each shipping provider, 
  // see that you must set flat methods too!
  // CONSTRAINT: Method's names MUST be UNIQUE
  // Script to create new shipping methods
  // http://demo.globant.com/~brovagnati/tools -> Shipping Method Generator
 
  $mc_shipping_methods = array(
                        'usps' => array(
                                    'domestic_types' =>
                                      array(
                                          'EXPRESS' => 'Express Mail',
                                          'FIRST CLASS' => 'First-Class Mail',
                                          'PRIORITY' => 'Priority Mail',
                                          'PARCEL' => 'Parcel Post',
                                          'BPM' => 'Bound Printed Material',
                                          'LIBRARY' => 'Library'
                                           ),
                                    'international_types' =>
                                      array(
                                          'Global Express Guaranteed' => 'Global Express Guaranteed (1 - 3 days)',
                                          'Global Express Guaranteed Non-Document Rectangular' => 'Global Express Guaranteed Non-Document Rectangular (1 - 3 days)',
                                          'Global Express Guaranteed Non-Document Non-Rectangular' => 'Global Express Guaranteed Non-Document Non-Rectangular (1 - 3 days)',
                                          'Express Mail International (EMS)' => 'Express Mail International (EMS) (3 - 5 days)',
                                          'Express Mail International (EMS) Flat Rate Envelope' => 'Express Mail International (EMS) Flat Rate Envelope (3 - 5 days)',
                                          'Priority Mail International' => 'Priority Mail International (6 - 10 days)',
                                          'Priority Mail International Flat Rate Box' => 'Priority Mail International Flat Rate Box (6 - 10 days)',
                                          'Priority Mail International Flat Rate Envelope' => 'Priority Mail International Flat Rate Envelope',
                                          'First-Class Mail International' => 'First-Class Mail International',
                                          
                                          ),                                        
                                        ),

                        'fedexexpress' => array(
                                    'domestic_types' =>
                                      array(
                                          '06' => 'FedEx First Overnight',
                                          '01' => 'FedEx Priority Overnight',
                                          '05' => 'FedEx Standard Overnight',
                                          '03' => 'FedEx 2Day',
                                          '20' => 'FedEx Express Saver',

                                           ),

                                    'international_types' =>
                                      array(
                                          '01' => 'FedEx International Priority',
                                          '03' => 'FedEx International Economy',

                                           ),
                                        ),
                        'fedexground' => array(
                                    'domestic_types' =>
                                      array(
                                          '90' => 'FedEx Home Delivery',

                                           ),

                                    'international_types' =>
                                      array(

                                           ),
                                        ),
                        'fedex1' => array(
                                    'domestic_types' =>
                                      array(
                                          '01' => 'Priority (by 10:30AM, later for rural)',
                                          '03' => '2 Day Air',
                                          '05' => 'Standard Overnight (by 3PM, later for rural)',
                                          '06' => 'First Overnight',
                                          '20' => 'Express Saver (3 Day)',
                                          '90' => 'Home Delivery',
                                          '92' => 'Ground Service'
                                           ),

                                    'international_types' =>
                                      array(
                                          '01' => 'International Priority (1-3 Days)',
                                          '03' => 'International Economy (4-5 Days)',
                                          '06' => 'International First',
                                          '90' => 'International Home Delivery',
                                          '92' => 'International Ground Service'
                                           ),
                                        ),
                        'ups' => array(
                                    'domestic_types' =>
                                      array(
                                          '1DM' => 'Next Day Air Early AM',
                                          '1DML' => 'Next Day Air Early AM Letter',
                                          '1DA' => 'Next Day Air',
                                          '1DAL' => 'Next Day Air Letter',
                                          '1DAPI' => 'Next Day Air Intra (Puerto Rico)',
                                          '1DP' => 'Next Day Air Saver',
                                          '1DPL' => 'Next Day Air Saver Letter',
                                          '2DM' => '2nd Day Air AM',
                                          '2DML' => '2nd Day Air AM Letter',
                                          '2DA' => '2nd Day Air',
                                          '2DAL' => '2nd Day Air Letter',
                                          '3DS' => '3 Day Select',
                                          'GND' => 'Ground',
                                          'GNDCOM' => 'Ground Commercial',
                                          'GNDRES' => 'Ground Residential',
                                          'STD' => 'Canada Standard',
                                          'XPR' => 'Worldwide Express',
                                          'XPRL' => 'worldwide Express Letter',
                                          'XDM' => 'Worldwide Express Plus',
                                          'XDML' => 'Worldwide Express Plus Letter',
                                          'XPD' => 'Worldwide Expedited'
                                           ),

                                    'international_types' =>
                                      array(

                                           ),
                                        ),
                        'zones' => array(
                                    'domestic_types' =>
                                      array(
                                          'zones' => 'Zones Rates'
                                           ),

                                    'international_types' =>
                                      array(
                                          'zones' => 'Zones Rates intl'
                                           ),
                                        ),
                        'freeoptions' => array(
                                    'domestic_types' =>
                                      array(
                                          'freeoptions' => 'Free Options'
                                           ),

                                    'international_types' =>
                                      array(
                                          'freeoptions' => 'Free Options intl'
                                           ),
                                        ),
                        'freeshipper' => array(
                                    'domestic_types' =>
                                      array(
                                          'freeshipper' => 'Free Shipper'
                                           ),

                                    'international_types' =>
                                      array(
                                          'freeshipper' => 'Free Shipper intl'
                                           ),
                                        ),
                        'perweightunit' => array(
                                    'domestic_types' =>
                                      array(
                                          'perweightunit' => 'Perweight Unit'
                                           ),

                                    'international_types' =>
                                      array(
                                          'perweightunit' => 'Perweight Unit intl'
                                           ),
                                        ),
                        'storepickup' => array(
                                    'domestic_types' =>
                                      array(
                                          'storepickup' => 'Store Pickup'
                                           ),

                                    'international_types' =>
                                      array(
                                          'storepickup' => 'Store Pickup intl'
                                           ),
                                        ),
                        'flat' => array(
                                    'domestic_types' =>
                                      array(
                                          'flat' => 'Flat Rate Per Order'
                                           ),

                                    'international_types' =>
                                      array(
                                          'flat' => 'Flat Rate Per Order intl'
                                           ),
                                        ),
                        'item' => array(
                                    'domestic_types' =>
                                      array(
                                          'item' => 'Flat Rate Per Item'
                                           ),

                                    'international_types' =>
                                      array(
                                          'item' => 'Flat Rate Per Item intl'
                                           ),
                                        ),
                        'table' => array(
                                    'domestic_types' =>
                                      array(
                                          'table' => 'Vary by Weight/Price'
                                           ),

                                    'international_types' =>
                                      array(
                                          'table' => 'Vary by Weight/Price intl'
                                           ),
                                        ),
                        'itemnational' => array(
                                    'domestic_types' =>
                                      array(
                                          'itemnational' => 'Item National',

                                           ),

                                    'international_types' =>
                                      array(

                                           ),
                                        ),
                        'iteminternational' => array(
                                    'domestic_types' =>
                                      array(

                                           ),

                                    'international_types' =>
                                      array(
                                          'iteminternational' => 'Item International',

                                           ),
                                        ),                                        
                                  );

  $mc_shipping_methods_names = array( 
                                         'usps' => 'USPS',
                                         'fedex1' => 'FedEx',
                                         'ups' => 'UPS',
                                         'zones' => 'Zones',
                                         'fedexexpress' => 'Fedex Express',
                                         'fedexground' => 'Fedex Ground',
                                         'freeoptions' => 'Free Options',
                                         'freeshipper' => 'Free Shipper',
                                         'perweightunit' => 'Perweight Unit',
                                         'storepickup' => 'Store Pickup',
                                         'flat' => 'Flat Rate',
                                         'item' => 'Item',
                                         'table' => 'Table',
                                         'itemnational' => 'Per Item National',
                                         'iteminternational' => 'Per Item International',
                                        );  
?>
